### PR TITLE
Add launcher to launch parental controls app

### PR DESCRIPTION
**Pull request by uajain**
Refresh implementation of https://github.com/endlessm/gnome-control-center/pull/206

https://phabricator.endlessm.com/T29440

___ 

Depends on meson build changes in #212 #211 (now merged)

### CODE REVIEW

File: panels/user-accounts/cc-user-panel.c [hunk 1]
```diff
@@ -1122,6 +1158,12 @@ activate_row (GtkListBox *box, GtkListBoxRow *row, CcUserPanel *self)
         } else if (row == self->last_login_row) {
                 show_history (self);
         }
+
+#ifdef HAVE_MALCONTENT
+        if (row == self->parental_controls_row) {
+		spawn_malcontent_control (self);
+	}
```
Comment (pwithnall) [inline panels/user-accounts/cc-user-panel.c]: Nitpick: Braces look misaligned here

File: panels/user-accounts/cc-user-panel.c [hunk 2]
```diff
@@ -859,6 +868,19 @@ show_user (ActUser *user, CcUserPanel *self)
...
 
+#ifdef HAVE_MALCONTENT
+        /* Parental Controls: Unavailable if user is admin */
+        if (act_user_get_account_type (user) == ACT_USER_ACCOUNT_TYPE_ADMINISTRATOR) {
+                gtk_widget_hide (GTK_WIDGET (self->parental_control_go_next));
+                gtk_label_set_text (self->parental_controls_button_label, _("Unavailable"));
+        } else {
+                gtk_widget_show (GTK_WIDGET (self->parental_control_go_next));
+                /* TODO: The label below should read "Enabled" or "Disabled" for parental-controls.
+                   We need to support querying that via a API in malcontent */
```
Comment (pwithnall) [inline panels/user-accounts/cc-user-panel.c]: (Reminder that this needs doing.)

File: panels/user-accounts/cc-user-panel.c [hunk 3]
```diff
@@ -795,6 +808,39 @@ get_autologin_possible (ActUser *user)
...
+	        return FALSE;
+        }
+
+        manager = mct_manager_new (system_bus);
+        app_filter = mct_manager_get_app_filter (manager,
+                                                 act_user_get_uid (user),
+                                                 MCT_GET_APP_FILTER_FLAGS_NONE,
+                                                 NULL,
+                                                 &error);
+        if (error) {
```
Comment (pwithnall) [inline panels/user-accounts/cc-user-panel.c]: Probably want to ignore (don’t print a warning for) `MCT_MANAGER_ERROR_DISABLED` which is emitted when parental controls are globally disabled (e.g. due to not being installed properly).

Comment (uajain) [reply panels/user-accounts/cc-user-panel.c]: Addressed (diff below for easier review)

```diff
diff --git a/panels/user-accounts/cc-user-panel.c b/panels/user-accounts/cc-user-panel.c
index b2c4303d5..fb7d242e5 100644
--- a/panels/user-accounts/cc-user-panel.c
+++ b/panels/user-accounts/cc-user-panel.c
@@ -831,9 +831,11 @@ is_parental_controls_enabled_for_user (ActUser *user)
                                                  NULL,
                                                  &error);
         if (error) {
-                g_warning ("Error retrieving app filter for user %s: %s",
-                           act_user_get_user_name (user),
-                           error->message);
+                if (!g_error_matches (error, MCT_MANAGER_ERROR, MCT_MANAGER_ERROR_DISABLED))
+                        g_warning ("Error retrieving app filter for user %s: %s",
+                                   act_user_get_user_name (user),
+                                   error->message);
+
                 return FALSE;
         }
```

**Pull request merged**